src/lib/src/component/navigation/navigation.component.ts

```diff
@@ -224,48 +230,21 @@ export class NavigationComponent implements OnInit, OnDestroy {
     }
   }
 
-  private setTestStatus(element: WorkspaceElement, status: string): void {
+  private setTestStatus(elementPath: string, status: string): void {
     switch (status) {
-      case 'RUNNING': element.state = ElementState.Running; break;
-      case 'SUCCESS': element.state = ElementState.LastRunSuccessful; break;
-      case 'FAILED': element.state = ElementState.LastRunFailed; break;
-      case 'IDLE': default: element.state = ElementState.Idle;
-    }
-  }
-
-  private expandNode(element: WorkspaceElement): void {
-    if (element !== null && element.type === ElementType.Folder) {
-      this.workspace.setExpanded(element.path, true);
-    }
-  }
-
-  private collapseNode(element: WorkspaceElement): void {
-    if (element !== null && element.type === ElementType.Folder) {
-      this.workspace.setExpanded(element.path, false);
-    }
-  }
-
-  private selectPredecessor(element: WorkspaceElement): void {
-    let predecessor = this.workspace.previousVisible(element);
-    if (predecessor != null) {
-      this.workspace.setSelected(predecessor.path);
-      this.changeDetectorRef.detectChanges();
-    }
-  }
-
-  private selectSuccessor(element: WorkspaceElement): void {
-    let successor = this.workspace.nextVisible(element);
-    if (successor != null) {
-      this.workspace.setSelected(successor.path);
-      this.changeDetectorRef.detectChanges();
+      case 'RUNNING': this.workspace.setTestStatus(elementPath, ElementState.Running); break;
+      case 'SUCCESS': this.workspace.setTestStatus(elementPath, ElementState.LastRunSuccessful); break;
+      case 'FAILED': this.workspace.setTestStatus(elementPath, ElementState.LastRunFailed); break;
+      case 'IDLE': default: this.workspace.setTestStatus(elementPath, ElementState.Idle);
     }
   }
 
-  private openFile(element: WorkspaceElement): void {
-    if (element !== null && element.type === ElementType.File) {
+  private openFile(elementPath: string): void {
+    const elementInfo = this.getWorkspace().getElement(elementPath);
```

Comment:
why 'elementInfo'? getElement suggests otherwise

Reply:
right, that's because I forgot to rename the method…